File: 1442-Count-Triplets-That-Can-Form-Two-Arrays-of-Equal-XOR/main.go
func countTriplets(arr []int) int {
    
    n := len(arr)
    prefix := make([]int, n+1)

    for i := range n {
        prefix[i + 1] = prefix[i] ^ arr[i]
    }

    var count int = 0
    for j := range n {
        for k := j + 1; k < n; k++{
            if prefix[j] == prefix[k+1]{
                count += ( k - j)
            }
        }
    }

    return count
}

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/main.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        
        n = len(arr)
        prefix = [0] * (n + 1)

        for i in range(n):
            prefix[i+1] = prefix[i] ^ arr[i]

        count = 0
        for j in range(n):
            for k in range(j + 1, n):
                if prefix[j] == prefix[k + 1]:
                    count += (k - j)

        return count
